### DISCUSSION TITLE
能否让antd5的ConfigProvider控制token去改变antd4的less变量

### DISCUSSION THREAD

**Discussion by YqxLzx** [ant-design/ant-design#46649] [Ideas]
可能有些人的项目有历史包袱，antd 4 和 5 版本共存了，但又没办法统一管理主题。

**Comment by yoyo837** [comment]
个人觉得不太可能提供这个支持，业务造成的历史包袱，业务自己解决。

**Reply by Wxh16144** [reply]
之前有做过一个表，但是这部分我也不知道怎么用。https://ant.design/docs/react/migrate-less-variables-cn#%E7%BB%84%E4%BB%B6%E5%8F%98%E9%87%8F

**Reply by yoyo837** [reply]
这个只能当文档用，没有功能

**Comment by MadCcc** [comment]
v4 无法通过运行时改变主题，只能通过 webpack 改 less 变量

**Reply by YqxLzx** [reply]
可是 antd4 能在运行时修改主题呀，这又是为什么

**Reply by MadCcc** [reply]
通过 js 运行时替换 css 文件，本质需要编译时生成多份 css 产物

**Reply by YqxLzx** [reply]
谢谢你的帮助 😄

**Comment by zombieJ** [comment]
less 是编译时的，不会动态跟着变化。
